### PR TITLE
[Console][HttpKernel] Handle new SHELL_VERBOSITY env var, also configures the default logger

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

On the CLI, the behavior of the new default logger is not nice: it's verbosity is controlled by `kernel.debug`, where I would expect to be able to control it by mean of verbosity.
To achieve this, I propose to handle a new `SHELL_VERBOSITY` env var, and use it to control both commands' verbosity, and the logger's log level.

### CODE REVIEW

**Comment by GromNaN** [comment]
The `Console\Application` class is doing a side effect on environment variables for usage in one implementation of the `Logger`.

Reading and writing global variables inside classes looks like a bad practice. This should be delegated to a config/context layer.

**Comment by nicolas-grekas** [comment]
@GromNaN it's not exactly for that, but to create an environment context that mirrors the `-vvv` verbosity flags. Then this context is leveraged by the logger yes, but this is independent (you could set the env var in a webserver context to control the logger's verbosity also.)
I don't have much better idea for now that would imply a pile of code we don't need.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by alexislefebvre** [comment]
TLDR: the value in `$output->getVerbosity()` is overwritten in the [`switch ($shellVerbosity = (int) getenv('SHELL_VERBOSITY')) {…}`](https://github.com/nicolas-grekas/symfony/blob/87bd741f2d4f6b66f03b5856d6bafa9fa7e7f6c5/src/Symfony/Component/Console/Application.php#L898) block added in this PR, is it an unexpected site effect?

---

It looks like this PR overwrites the verbosity level set by calling `setVerbosity()`. Maybe this wasn't a good idea though.

Anyway, here is how we used it in LiipFunctionalTestBundle:

1. we created an `Application`: https://github.com/liip/LiipFunctionalTestBundle/blob/bf909887fbd3aac10215efeb6f6287c72afc568d/Test/WebTestCase.php#L130
2. we instantiated a `StreamOutput` with the verbosity level: https://github.com/liip/LiipFunctionalTestBundle/blob/bf909887fbd3aac10215efeb6f6287c72afc568d/Test/WebTestCase.php#L146 (so we expect `StreamOutput->verbosity` to contain the expected verbosity value)
3. we called `Application::run()`: https://github.com/liip/LiipFunctionalTestBundle/blob/bf909887fbd3aac10215efeb6f6287c72afc568d/Test/WebTestCase.php#L148
4. this last method calls `configureIO` that has been changed in this PR: https://github.com/symfony/symfony/pull/24425/files#diff-38eebdda7d6ebfcb7405b91167f05c9c

During my own tests, `getenv('SHELL_VERBOSITY')` is equal to `3` (I'm still wondering why). So `$output->setVerbosity(OutputInterface::VERBOSITY_DEBUG);` is called and the value in `$output->getVerbosity();` is overwritten.

**Comment by xabbuh** [comment]
> During my own tests, `getenv('SHELL_VERBOSITY')` is equal to `3` (I'm still wondering why).

That's probably because of the `Kernel` class setting it explicitly if it wasn't set before: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/Kernel.php#L115-L119

For the rest, I suggest to open a new issue if you feel something is not working as expected.

**Comment by lisachenko** [comment]
Unexpected result of this PR that CLI commands in debug mode now do not write debug and info logs, because of default value `LogLevel::WARNING`. I think that for debug mode verbosity should be debug to keep logging all debug lines.
